lib/engine/game/g_1854/step/merge_minors.rb

```diff
@@ -0,0 +1,114 @@
+            return [] if target_corporations.empty?
+            return [] if @merge_target
+
+            actions = []
+            actions << 'merge' if @game.minor_mergers_allowed?
+            actions << 'pass' if @game.minor_mergers_allowed? && !@game.minor_mergers_required?
+            actions
+          end
+
+          def blocking?
+            @game.mergeable?(current_entity)
+          end
+
+          def active?
+            return false if @game.open_minors.empty?
+
+            @game.minor_mergers_allowed? || @game.minor_mergers_required?
+          end
+
+          def setup
+            @mergeable_ = {}
+            @merging = nil
+            @merge_target = nil
+          end
+
+          def auto_actions(entity)
+            return super if @merging
+
+            return [Engine::Action::Pass.new(entity)] if mergeable_candidates(entity).empty?
+
+            super
+          end
+
+          def merge_name(entity = nil)
+            return "Form #{entity&.name}" if @merging
+
+            "Merge with #{entity&.name}"
+          end
+
+          # corps available as a merge target
+          def target_corporations
+            @game.corporations.select { |c| @game.merge_target?(c) }
+          end
+
+          def description
+            return 'Choose corporation to form' if @merging
+
+            'Merge'
+          end
+
+          def process_merge(action)
+            if @merging
+              @merge_target = action.corporation
+              raise GameError, "#{@merge_target.name} is not available to merge into" unless @game.merge_target?(@merge_target)
+
+              @game.merge_minors_into_lokalbahn(action.entity, @merging, @merge_target)
+            else
+              other = action.minor
+
+              raise GameError, "#{other.name} is the wrong corporation type" if other.type != action.entity.type
+              raise GameError, "#{other.name} is not available to merge with" unless @game.mergeable?(other)
+
+              @merging = other
+
+              @log << "#{@game.formatted_minor_name(action.entity)} and #{@game.formatted_minor_name(other)} selected to merge."
+            end
+          end
+
+          def mergeable_type(corporation)
+            return "Corporations that #{corporation.name} and #{@merging.name} can be merged into to form" if @merging
+
+            "Minors that can merge with #{corporation.name}"
+          end
+
+          def mergeable_candidates(corporation)
+            # # Mergeable candidates must be connected by track and not through a regional border
+            # # They must be the same type (major/minor)
+            # parts = @game.token_graph_for_entity(corporation).connected_nodes(corporation).keys
+            # mergeable = parts.select { |n| n.city? && !n.pass? }.flat_map { |c| c.tokens.compact.map(&:corporation) }
+            @mergeable_[corporation] ||=
+              @game.minors.uniq.select { |c| c != corporation && @game.mergeable?(c) }
```

Comment:
What does `@game.mergeable?` do?

Reply:
https://github.com/andrewzwicky/18xx/blob/80f2537d5a4c495dc9f4612e7f5b6cb913236bbb/lib/engine/game/g_1854/game.rb#L279-L281

Reply:
I don't know this bit of the codebase, but do you need the `@mergeable_` hash for this game? You might just need to return all non-closed minors.

Reply:
Probably not, it looks like a method to save time recomputing expensive graph calculations in other features.